### DISCUSSION TITLE
A build with roughly sourced parts

### DISCUSSION THREAD

**Discussion by kpishere** [slu4coder/Minimal-64x4-Home-Computer#23] [Q&A]
So, here is my new build. On reset it is using 180 mA, after reset it is using 220 mA, so within spec there.

All parts were from Aliexpress and I did test all except for the counters and parallel shift registers (74hc161 and 74hc166).  5 out of 10 of the 161's were bad (you can see those spots replaced on board where small chips have sockets)!

![IMG_0618](https://github.com/user-attachments/assets/8e364fec-6fa7-4bf1-9845-7bb158dd1e11)

Here is the screen output when in reset.

![IMG_0616](https://github.com/user-attachments/assets/9e122f7c-8f78-48a8-99f0-12cee560a720)

Here is the screen output when running.

![IMG_0617](https://github.com/user-attachments/assets/04bb79d7-71d5-43c2-a36a-bc5d06990448)

Is it just monitor sync issue?  Or what?  Try another shift register?

**Comment by kpishere** [comment]
Got some more sync, info.  Unfortunately, can't set in interface of monitor.

![syncinfo](https://github.com/user-attachments/assets/a5a1e3b8-4bf7-4c5a-b50b-45aa52177f86)

**Comment by kpishere** [comment]
So that screen shot above suggests to me that there is no effective blanking happening.  I've replaced 74HC166 with a genuine Phillips brand and the screen looks more regular.  This looks like it is displaying what is in video ram or maybe a single location of video ram.  At this point, I'm gonna replace all 74HC161 chips with genuine ones.  

This is the image after replacing above ..
![IMG_0620](https://github.com/user-attachments/assets/4a1aff99-9bb3-4c4d-ab35-4ff1b4d2215a)

**Comment by kpishere** [comment]
Got some 74HC161 chips locally of new old stock from Motorola.  Boy oh boy, what a difference.  The timing signals are now rock solid. Previously, there was 10-15% jitter and variation over input voltage between 4.7 to 5.0.  Now, there is none.  Still though, similar output to above, will have to change some more chips.  Yep, big mistake not to order from mouser or digi-key.  These parts are just too jelly-bean to order from Aliexpress reliably.

**Comment by kpishere** [comment]
Video signals are nice but still, something not running.  Below, I've traced all signal lines involved in first instruction, MIW.

Any insight, much appreciated.  Any way to look at these signals from an emulator?

![Screen Shot 2024-08-02 at 8 52 41 PM](https://github.com/user-attachments/assets/1c949d66-98f0-4237-8825-49fae99c3031)

**Reply by kpishere** [reply]
Zooming out from the above, it looks like it is getting stuck at copying OS in first few instructions .. 

![Screen Shot 2024-08-02 at 9 24 20 PM](https://github.com/user-attachments/assets/f68fbd1d-d7d8-4967-a13f-0f3c63bd81e6)

**Reply by kpishere** [reply]
Breaking down each statement, something does not look right at CIB FNE and CIB FCC.

![Screen Shot 2024-08-02 at 9 47 07 PM](https://github.com/user-attachments/assets/309b91d3-ba09-4e25-8ec7-ccd84b823deb)

**Reply by kpishere** [reply]
The last three statements above are not as shown, they are .. from the first CIB, FNE, then LSR, STR, INW, INW, ...
Following all the way to 255 copies, there is no pattern change at 13.763 ms.  A carry isn't working?  So it is stuck copying the first 256 bytes over and over.

**Comment by kpishere** [comment]
Dang, is hard to go further without some reference hardware that works from here.  So I bit the bullet and re-ordered all new chips from DigiKey.  All 74 series chips are Ti branded.  Low and behold, the thing works on the first go!  I did use the SST chips I'd bought from China, they were fine.

I'll go back and hook up the working one how under the logic analyzer and then compare the difference, see if I can spot which chips were no good in the non-working one.
![IMG_3685](https://github.com/user-attachments/assets/7c9c8e37-32d0-46f9-8e2f-cf6cf02a2556)

**Reply by nateo87** [reply]
I'm very anxious to see what you come up with comparing the logic between the "legit" chips and "brand x". Should be very interesting to see what exactly is easily tripped up.

**Reply by kpishere** [reply]
Well, here it is.  Actually, I'm quite surprised at all the differences, I'll have to take some time to digest this and the order of execution.
![Screen Shot 2024-08-25 at 9 00 41 AM](https://github.com/user-attachments/assets/64ec64af-98b9-40a2-8f2a-5ddb8b585ffd)

**Comment by slu4coder** [comment]
Hi kpisphere, sorry for not getting back to you. Life got in the way for almost 3 months at my end. I am glad to read that you got the Minimal 64x4 up and running all by yourself. And yes, you have repeated the experience of other builders with IC ordered from budged suppiers. Hard to tell what causes the issues with cheap IC copies. In one case I believe IC where sold als "HC" when in fact they were "LS".
